spec/models/poly_metadatum_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PolyMetadatum, type: :model do
+  subject(:test_metadatum) { build(:poly_metadatum) }
+
+  # Tests for validations
+  describe 'validations' do
+    it 'requires a key' do
+      expect(test_metadatum).to validate_presence_of(:key)
+    end
+
+    it 'requires a value' do
+      expect(test_metadatum).to validate_presence_of(:value)
+    end
+
+    it 'key must be unique' do
+      expect(test_metadatum).to validate_uniqueness_of(:key).scoped_to(:metadatable_id).case_insensitive
```

Comment:
Does this need to be scoped to metadatable_id and metadatable_type, otherwise if you have Request with id 52 and Receptacle with id 52, who both have a metadatum with key 'comment', you're in trouble?

Reply:
So I think maybe that test is wrong. There is a metadatable_type and a metadatable_id that in combination define the metadatable instance. So in your example one type should be 'requests' and the other 'receptacles' so they shouldn't clash.
Abdullah made the point that this would still be a confusing table to look at if we have many matching keys.

We could enforce unique key names with a join table with a unique index on key, and encourage a model specific prefix. 
e.g. with columns: `key, description, data_type, friendly name`
e.g. example: `'request_comment', 'Holds a text comment from a user', string, 'Comment'`

Reply:
Hmm. Getting this odd error
```

PolyMetadatum validations key must be unique for the metadatable
     Failure/Error: expect(test_metadatum).to validate_uniqueness_of(:key).scoped_to(%i[metadatable_type metadatable_id]).case_insensitive
     
     ActiveRecord::SubclassNotFound:
       Invalid single-table inheritance type: Request is not a subclass of Shoulda::Matchers::ActiveRecord::Uniqueness::TestModels::Requesu
     # ./spec/models/poly_metadatum_spec.rb:15:in `block (3 levels) in <top (required)>'
```

Reply:
Changed the scope to include both and modified test, seems to behave correctly

Reply:
I personally think it's OK - I think if you're reading it in the database you'd probably always do a join so that you have the metadatable type as a column, or you'd be filtering to a specific record or set of records all of the same type.

I like the idea of a 'friendly name' though - that's one of the things I dislike about 'custom metadata', as it looks rubbish in the UI displaying a key that's lowercase with underscores. (although iirc there's nothing preventing you making your key 'friendly')